Content.Shared/_DV/Light/SharedLightReactiveSystem.cs

```diff
@@ -77,33 +77,65 @@ public float GetLightLevelForPoint(EntityUid uid)
+            var radius = lightComp.Radius;
+            if (!lightComp.NetSyncEnabled)
+            {
+                // Try to use the GetLightEnergyEvent if we can't rely on it being network synced.
+                var lightEnergyEvnt = new GetLightEnergyEvent(uid);
+                RaiseLocalEvent(lightUid, lightEnergyEvnt);
+                if (!lightEnergyEvnt.Cancelled)
+                    continue; // No component claimed this light, skip.
+                energy = lightEnergyEvnt.LightEnergy;
+                radius = lightEnergyEvnt.LightRadius;
+            }
+
+            energy = MathF.Min(energy, 2f); // Clamp energy, to normalize strange values.
+
             // Ensure we're on the same grid as the light source
             if (_transform.GetMap(lightUid) != map)
                 continue;
 
             // Ensure we're within the light's radius.
             var lightPos = _transform.GetWorldPosition(lightUid);
             var sqrDistance = Vector2.DistanceSquared(pos, lightPos);
-            if (sqrDistance > lightComp.Radius * lightComp.Radius)
+            if (sqrDistance > radius * radius)
                 continue;
 
             if (sqrDistance < 0.01f)
             {
                 // If we're right on top of the light, just add its full energy value.
-                val += lightComp.Energy;
+                val += energy;
                 continue;
             }
 
             // Collision ray check from the entity to the light source
             var ray = new CollisionRay(pos, (lightPos - pos).Normalized(), (int)CollisionGroup.Opaque);
-            var hit = _physics.IntersectRay(_transform.GetMapId(uid), ray, MathF.Sqrt(sqrDistance) - 0.5f, returnOnFirstHit: true);
+            var hit = _physics.IntersectRay(_transform.GetMapId(uid), ray, MathF.Sqrt(sqrDistance) - 0.5f, ignoredEnt: lightUid, returnOnFirstHit: true);
             if (hit.Any() && hit.First().Distance != 0)
                 continue;
+
+            // Manual hack for cones.
+            if (lightComp.MaskPath == "/Textures/Effects/LightMasks/cone.png"){
+                var forward = _transform.GetWorldRotation(lightUid).RotateVec(new Vector2(0.0f, -1.0f));
+                energy *= MathF.Max(0f, Vector2.Dot((pos - lightPos).Normalized(), forward));
+            }
+
             // If we reach here, the light is unobstructed and within range, calculate a light value to add.
-            val += lightComp.Energy * (1.0f - sqrDistance / (lightComp.Radius * lightComp.Radius));
+            val += energy * (1.0f - sqrDistance / (radius * radius));
         }
 
 
         return val;
     }
 }
+
+/// <summary>
+/// Passed to unsync'd light sources to get the expected light energy.
+/// ONLY called when NetSync is not enabled. Otherwise, uses the light directly.
+/// </summary>
+public sealed class GetLightEnergyEvent(EntityUid Target) : CancellableEntityEventArgs
+{
+    public float LightEnergy = 0f;
+    public float LightRadius = 0f;
+}
```

Comment:
ByRefEvent + record struct + bool Handled instead of inheritance

you shouldnt have even used cancel to begin with it's the wrong word